lib/generateClass.js

```diff
@@ -205,6 +213,66 @@ function writeFromJson(def, cw) {
   cw.ln('return inst;');
 }
 
+/**
+ * Generates a JSON serializer for the element
+ * @param {DataElement} def - The definition of the SHR element to generate a serializer for
+ * @param {CodeWriter} cw - The CodeWriter instance to use during generation
+ * @private
+ */
+function writeToJson(def, cw) {
+  // If the element is an Entry, put those fields on the JSON object first
+  if (def.isEntry) {
+    cw.ln(`const inst = this._entryInfo.toJSON();`);
+    const url = `http://standardhealthrecord.org/spec/${def.identifier.namespace.replace('.', '/')}/${className(def.identifier.name)}`;
+    cw.ln(`inst['shr.base.EntryType'] = { 'Value' : '${url}' };`);
+  } else if (def.identifier.name !== 'EntryType') {
+    const url = `http://standardhealthrecord.org/spec/${def.identifier.namespace.replace('.', '/')}/${className(def.identifier.name)}`;
+    cw.ln(`const inst = { 'shr.base.EntryType': { 'Value' : '${url}' } };`);
+  } else {
+    cw.ln(`const inst = {};`);
+  }
+
+  if (def.value !== undefined) {
+    if (def.value instanceof ChoiceValue) {
+      // Choices get a special treatment
+      cw.ln(`inst['Value'] = typeof this.value.toJSON === 'function' ? this.value.toJSON() : this.value;`);
+    } else if (def.value instanceof IdentifiableValue && def.value.identifier.isPrimitive) {
+      cw.ln(`inst['Value'] = this.value;`);
+    } else {
+      generateAssignmentIfList(def.value.card, 'Value', 'value', cw);
+    }
+  }
+
+  for (const field of def.fields) {
+    if (!(field.identifier.name instanceof TBD)) {
+      if (field.identifier.isPrimitive) {
```

Comment:
Based on the grammar, it's impossible for any `def.fields` to be a primitive (only `def.value` can contain primitives).  So you shouldn't have to check if `field.identifier` is primitive -- it never will be.